packages/ui-react/src/components/LanguageMenu/LanguageMenu.tsx

```diff
@@ -13,14 +18,54 @@ type Props = {
 };
 
 const LanguageMenu = ({ onClick, languages, currentLanguage }: Props) => {
+  const [languageMenuOpen, setLanguageMenuOpen] = useState(false);
+
+  const handleOnBlur = () => setLanguageMenuOpen(false);
+  const handleOnFocus = () => setLanguageMenuOpen(true);
+  const handleMenuToggle = () => setLanguageMenuOpen(!languageMenuOpen);
+  const handleLanguageSelect = (event: MouseEvent<HTMLElement>, code: string) => {
+    event.preventDefault();
+    if (onClick) {
+      onClick(code);
+    }
+
+    setLanguageMenuOpen(false);
+  };
+
   return (
-    <ul className={styles.menuItems}>
-      {languages.map(({ code, displayName }) => (
-        <li key={code} className={classNames(styles.menuItem, { [styles.menuItemActive]: currentLanguage?.code === code })}>
-          <Link onClick={() => onClick?.(code)}>{displayName}</Link>
-        </li>
-      ))}
-    </ul>
+    <Fragment>
+      <IconButton
+        aria-controls="language-panel"
+        aria-expanded={languageMenuOpen}
+        className={classNames(styles.iconButton, styles.actionButton)}
+        aria-label={t('select_language')}
+        onClick={handleMenuToggle}
```

Comment:
By adding one more event listener, we make sure that if the icon is opened by pressing ENTER and you press SHIFT + TAB afterwards, the popup will close again. This will make the behaviour of the popup more consistent when you leave the popup (either leaving the first or last menu item).

```suggestion
        onClick={handleMenuToggle}
        onBlur={handleOnBlur}
```

Reply:
Agreed, but the handler doesn't need `on` in its name:
`onBlur={handleBlur}`

Reply:
- [x] Updated the function handlers to remove the on keyword.